src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotHideBaseClassMethods.cs

```diff
@@ -33,7 +33,7 @@ public sealed class DoNotHideBaseClassMethodsAnalyzer : DiagnosticAnalyzer
             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182143.aspx",
             customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX ? ImmutableArray.Create(Rule) : ImmutableArray<DiagnosticDescriptor>.Empty;
```

Comment:
Flag `DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX` will be false when building our analyzers VSIX.